### PR TITLE
style: change gcheap

### PR DESCRIPTION
**Pull request by songzhibin97**

### CODE REVIEW

**Review by doujiang24 — APPROVED** [review]

File: holmes.go [hunk 1]
```diff
@@ -117,31 +117,23 @@ func (h *Holmes) DisableMemDump() *Holmes {
...
-// it won't fit into tiny span since this struct contains point.
-type foo struct {
-	h *Holmes
-}
-
-func finalizerCallback(f *foo) {
+func finalizerCallback(h *Holmes) {
 	// register the finalizer again
-	runtime.SetFinalizer(f, finalizerCallback)
+	runtime.SetFinalizer(h, finalizerCallback)
```
Comment (doujiang24) [inline holmes.go:122]: oh, no, we should not `h` here since it may be referred in other place, it won't be GC sweep then.

**Review by doujiang24 — CHANGES_REQUESTED** [review]

File: holmes.go (hunk 1 above)
Comment (songzhibin97) [reply holmes.go:122]: oh,I may have misunderstood it.

File: holmes.go [hunk 2]
```diff
@@ -117,31 +118,31 @@ func (h *Holmes) DisableMemDump() *Holmes {
 	return h
 }
 
-// it won't fit into tiny span since this struct contains point.
```
Comment (doujiang24) [inline holmes.go:120]: this comment is also useful.

**Review by doujiang24 — COMMENTED** [review]
otherwise, LGTM.

**Review by doujiang24 — APPROVED** [review]

**Pull request merged**